src/sknnr/transformers/_constrained_transformer.py

```diff
@@ -16,11 +32,15 @@ def fit(self, X, y):
             ensure_min_features=2,
             ensure_min_samples=1,
         )
-        y = np.asarray(y)
-        if len(y.shape) < 2:
-            raise ValueError("`y` must be a 2D array.")
 
-        self.ordination_ = CCA(X, y)
+        method_cls = self.CONSTRAINED_METHODS.get(self.constrained_method)
+        if method_cls is None:
+            raise ValueError(
+                f"`method` must be one of {self.CONSTRAINED_METHODS.keys()}, not"
```

Comment:
```suggestion
                f"`method` must be one of {list(self.CONSTRAINED_METHODS.keys())}, not"
```

My bad for suggesting this originally! As is, it will print `method must be one of dict_keys(['cca', 'rda'])`.

Reply:
Thanks for the fix.  I've changed this.